### ISSUE TITLE
[Bug] train_autograd.py 中 ReduceLROnPlateau 使用 train_loss 而非 val 指標驅動學習率調降

### ISSUE THREAD

**Issue by s9213712** [s9213712/minicnn#50]
## 問題描述

`src/minicnn/training/train_autograd.py` 在呼叫 `ReduceLROnPlateau.step()` 時
傳入 `row['train_loss']`（訓練集 loss）而非驗證集指標，
導致學習率調降邏輯監控錯誤的對象。

## 問題程式碼

```python
# training/train_autograd.py (~line 217)
if scheduler is not None:
    if isinstance(scheduler, ReduceLROnPlateau):
        scheduler.step(metric=row['train_loss'])  # ❌ 應使用 val_loss 或 val_acc
    else:
        scheduler.step()
```

## 正確行為

`ReduceLROnPlateau` 的設計目的是「當驗證指標停止改善時降低學習率」。
使用 training loss 會造成：
- 訓練集仍在 overfitting 時 LR 已被誤降
- 或訓練集 loss 持續下降（即使 val 已停滯），LR 永遠不降

## 對照

torch backend 的 flex/\_training\_run.py 正確地使用 `val_metrics['loss']`：
```python
def step_scheduler(torch, scheduler, *, val_metrics):
    if isinstance(scheduler, torch.optim.lr_scheduler.ReduceLROnPlateau):
        scheduler.step(val_metrics['loss'])  # ✓ 正確
```

## 根本原因

autograd 訓練路徑的 `_accuracy()` 函數只回傳 accuracy，不計算 val\_loss。
應補充 val\_loss 計算，或改用 val\_acc 搭配 `mode='max'`。

## 受影響路徑

- `src/minicnn/training/train_autograd.py`: L217
- `src/minicnn/schedulers/plateau.py`: 無 `mode='min'/'max'` 參數，預設只支援 minimize

## 嚴重性

邏輯錯誤，影響所有以 autograd backend + plateau scheduler 執行的訓練。
